libs/core/src/main/java/org/elasticsearch/core/Strings.java

```diff
@@ -21,8 +21,14 @@ public class Strings {
      *
      * This method calls {@link String#format(Locale, String, Object...)}
      * with Locale.ROOT
+     * If format is incorrect the function will return format without populating
+     * its variable placeholders.
      */
     public static String format(String format, Object... args) {
-        return String.format(Locale.ROOT, format, args);
+        try {
+            return String.format(Locale.ROOT, format, args);
+        } catch (Throwable e) {
+            return format;
```

Comment:
Can we not fail silently? We should somehow capture the actual problem, embedding it in the message. Also, add an assert here so that if we hit this in tests we can fail the test.

Reply:
how should we capture it? Maybe we should log it (with additional Strings's logger)

Reply:
Yes, logging sounds good.

Reply:
Actually, not sure we can do logging, because this is in the core lib? It doesn't have any deps. Perhaps we could embed it in the message?

Reply:
I am not sure.. This would be a good solution if we only use this in logging (we do, but the method is in core with no indication that this is only for logging).
I wonder if just returning the original pattern is not good enough. This is what log4j is doing when a pattern has more place holders. 
I wonder if we could rather just try to use some static analysis to detect this
https://rules.sonarsource.com/java/RSPEC-2275 
this looks promising but would have to be adopted to our needs

Reply:
I'm OK with leaving it as is if it matches what log4j does.

Reply:
I raised an issue to look for static analysis solution this https://github.com/elastic/elasticsearch/issues/87166